schema.sql

```diff
@@ -0,0 +1,159 @@
+
+--Table Network With The Whole Users Network/Connections:
+CREATE TABLE IF NOT EXISTS user_network (
+    id serial,
+    facebook_id varchar(16),
+    met_with varchar(16),
+    latitude numeric,
+    longitude numeric,
+    info varchar(500),
+    timestamp timestamp without time zone default (now() at time zone 'utc'),
+    unix_ts int default extract(epoch from now())
+);
+CREATE INDEX IF NOT EXISTS user_network_id0 ON user_network (facebook_id);
+CREATE INDEX IF NOT EXISTS user_network_id1 ON user_network (met_with);
+--DROP TABLE IF EXISTS network CASCADE;
+
+--View To Easily Get Latest User Health Status For Each User:
+DO $$
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'latest_status') THEN
+        CREATE VIEW latest_status AS
+            SELECT a.user_id, a.status, a.symptoms, a.timestamp
+            FROM history a WHERE NOT EXISTS (
+                SELECT 1 FROM history b
+                WHERE a.user_id = b.user_id
+                  AND a.timestamp < b.timestamp);
+    END IF;
+END$$;
+--DROP VIEW latest_status;
+
+--Dummy Data For Test Proposes
+DELETE FROM users where id in ('1','2','3');
+DELETE FROM history where user_id in ('1','2','3');
+DELETE FROM network where user_id in ('1','2','3');
+INSERT INTO users(id, hash, year, postalcode, ip, info) values('1', '1', 1, '1000-100', null, null);
+INSERT INTO users(id, hash, year, postalcode, ip, info) values('2', '2', 1, '1000-100', null, null);
+INSERT INTO users(id, hash, year, postalcode, ip, info) values('3', '3', 1, '1000-100', null, null);
+INSERT INTO network(user_id, facebook_id, name, email, url) values('1', '1', 'name1', 'email1@email.com', 'http://url1.com');
+INSERT INTO network(user_id, facebook_id, name, email, url) values('2', '2', 'name2', 'email2@email.com', 'http://url2.com');
+INSERT INTO network(user_id, facebook_id, name, email, url) values('3', '3', 'name3', 'email3@email.com', 'http://url3.com');
+INSERT INTO history(user_id, status, symptoms) values('1', 'normal', False);
+INSERT INTO history(user_id, status, symptoms) values('1', 'infected', True);
+INSERT INTO history(user_id, status, symptoms) values('1', 'recovered', False);
+INSERT INTO history(user_id, status, symptoms) values('2', 'normal', False);
+INSERT INTO history(user_id, status, symptoms) values('2', 'infected', True);
+INSERT INTO history(user_id, status, symptoms) values('3', 'normal', False);
+INSERT INTO user_network(facebook_id, met_with) values('1', '2');
+INSERT INTO user_network(facebook_id, met_with) values('1', '3');
+INSERT INTO user_network(facebook_id, met_with) values('2', '1');
+INSERT INTO user_network(facebook_id, met_with) values('2', '3');
+INSERT INTO user_network(facebook_id, met_with) values('3', '1');
+INSERT INTO user_network(facebook_id, met_with) values('3', '2');
+
+--Get Current Status Count:
+SELECT status, count(1) FROM latest_status GROUP BY status;
+
+--Dummy Table With Column Type Examples
+CREATE TABLE IF NOT EXISTS example (
```

Comment:
Is this necessary?

Reply:
nope, it is just there to show some datatypes in case someone wants to add/change something and doesn't know the sintax, we can remove it.